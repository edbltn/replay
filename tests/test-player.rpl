strat {A, B};

Strategy s = Strategy[2|4|{  state0: A, (A,A) -> state1
                                      | (A,B) -> state2;
                             state1: A, (_,_) -> state2;
                             state2: B, (B,_) -> state1
                                      | (B,A) -> state0;}];
Player p = Player[s|1.0];

println(p.payoff);
println(p.rounds);
println(p.state);
println(p.strategy);
