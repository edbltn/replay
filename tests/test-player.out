0
0
0
State 0: play 0
( 0 0 ) -> state 1
( 0 1 ) -> state 2
( 1 0 ) -> state 0
( 1 1 ) -> state 0
State 1: play 0
( 0 0 ) -> state 2
( 0 1 ) -> state 2
( 1 0 ) -> state 2
( 1 1 ) -> state 2
State 2: play 1
( 0 0 ) -> state 2
( 0 1 ) -> state 2
( 1 0 ) -> state 0
( 1 1 ) -> state 1
State 3: play 0
( 0 0 ) -> state 3
( 0 1 ) -> state 3
( 1 0 ) -> state 3
( 1 1 ) -> state 3

